### PR TITLE
[Doc] API Key deletion settings

### PR DESCRIPTION
**Pull request by albertzaharovits**
Documents API Key delete settings.

### CODE REVIEW

File: docs/reference/settings/security-settings.asciidoc [hunk 1]
```diff
@@ -202,11 +202,6 @@ You can set the following API key service settings in
 (<<static-cluster-setting,Static>>)
 Set to `false` to disable the built-in API key service. Defaults to `true`.
 
-`xpack.security.authc.api_key.hashing.algorithm`::
-(<<static-cluster-setting,Static>>)
-Specifies the hashing algorithm that is used for securing API key credentials.
-See <<password-hashing-algorithms>>. Defaults to `pbkdf2`.
-
```
Comment (albertzaharovits) [inline docs/reference/settings/security-settings.asciidoc:209]: moved below, for alphabetic ordering.

File: docs/reference/settings/security-settings.asciidoc [hunk 2]
```diff
@@ -224,6 +219,31 @@ The hashing algorithm that is used for the
...
 Defaults to `ssha256`.
 
+[[api-key-service-settings-delete-retention-period]]
+`xpack.security.authc.api_key.delete.retention_period`::
+(<<dynamic-cluster-setting,Dynamic>>)
+Invalidated or expired API keys older than the retention period are eligible for deletion.
+Defaults to `7d`.
+Note that large real-time clock inaccuracies across the cluster nodes can cause problems
+with evaluating the API key retention period (in the likely case that the node performing
+the invalidation is different from the one performing the deletion).
```
Comment (slobodanadamovic) [inline docs/reference/settings/security-settings.asciidoc:229]:
```suggestion

--
NOTE:  The large real-time clock inaccuracies across the cluster nodes can cause problems
with evaluating the API key retention period (in the likely case that the node performing
the invalidation is different from the one performing the deletion).

--
```

File: docs/reference/settings/security-settings.asciidoc [hunk 3]
```diff
@@ -224,6 +219,31 @@ The hashing algorithm that is used for the
...
+
+`xpack.security.authc.api_key.delete.interval`::
+(<<dynamic-cluster-setting,Dynamic>>, Expert)
+Cluster nodes schedule the automatic deletion of invalidated or expired API keys
+that are older than the retention period.
+This setting controls the minimum time interval between two such deletion jobs.
+Defaults to `24h`.
```
[the remark this answers is not in the document]
Comment (slobodanadamovic) [reply docs/reference/settings/security-settings.asciidoc:236]: I'm okay with documenting it is as is. Optionally, we could also add your explanation as a note.

File: docs/reference/settings/security-settings.asciidoc [hunk 4]
```diff
@@ -224,6 +219,31 @@ The hashing algorithm that is used for the
...
+
+`xpack.security.authc.api_key.delete.timeout`::
+(<<static-cluster-setting,Static>>, Expert)
+Sets the timeout of the internal search and delete call.
```
[the remark this answers is not in the document]
Comment (slobodanadamovic) [reply docs/reference/settings/security-settings.asciidoc:240]: I see your point and I don't have big preference to document it. I'll leave this decision to your discretion.

File: docs/reference/settings/security-settings.asciidoc (hunk 2 above)
Comment (slobodanadamovic) [reply docs/reference/settings/security-settings.asciidoc:229]: nit: we can wrap this as a note.

**Review by slobodanadamovic — APPROVED** [review]
LGTM 

I left few non-blocking comment.

File: docs/reference/settings/security-settings.asciidoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (albertzaharovits) [reply docs/reference/settings/security-settings.asciidoc:240]: @gwbrown Any preference here? Should we keep this msg vague and professional looking, but not accurate, or should we document the janky timeout working? Or not document the timeout setting at all?

**Comment by jakelandis** [comment]
https://elasticsearch_bk_105392.docs-preview.app.elstc.co/diff is the correct preview link

File: docs/reference/settings/security-settings.asciidoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (gwbrown) [reply docs/reference/settings/security-settings.asciidoc:240]: I think my opinion is to leave it documented, and add a note with the explanation. I understand the impulse not to - it is indeed very low level and might be confusing to customers - but we're also at a point where having knowledge like that silo'd into a few people on the ES team is a bit dangerous. At the very least I think we should do something vague like:

> This setting controls the time interval between two such deletion jobs. Note that this setting is approximate and deletion jobs may be triggered more often than this interval.

I could be convinced to remove this from the docs, but I'd want a good reason not to err on the side of writing everything down.

File: docs/reference/settings/security-settings.asciidoc [hunk 5]
```diff
@@ -224,6 +219,35 @@ The hashing algorithm that is used for the
...
+[[api-key-service-settings-delete-retention-period]]
+`xpack.security.authc.api_key.delete.retention_period`::
+(<<dynamic-cluster-setting,Dynamic>>)
+Invalidated or expired API keys older than the retention period are eligible for deletion.
+Defaults to `7d`.
+
+--
+NOTE:  The large real-time clock inaccuracies across the cluster nodes can cause problems
+with evaluating the API key retention period (in the likely case that the node performing
+the invalidation is different from the one performing the deletion).
```
Comment (gwbrown) [inline docs/reference/settings/security-settings.asciidoc:231]: Nitpicky changes to the wording here:

```suggestion
NOTE:  Large real-time clock inconsistency across cluster nodes can cause problems
with evaluating the API key retention period. That is, if the clock on the node 
invalidating the API key is significantly different than the one performing the deletion,
the key may be retained for longer or shorter than the configured retention period.
```

I think that's a bit clearer? Writing this well is tricky.

File: docs/reference/settings/security-settings.asciidoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (gwbrown) [reply docs/reference/settings/security-settings.asciidoc:244]: Unless we have a good reason to not document it, I'm in favor of erring on the side of documenting it. As it stands this is the reference for everyone inside and outside of Elastic for Elasticsearch and I'd rather someone who's looking be able to find it than have to wake people up until we find someone who knows about this in an emergency.

File: docs/reference/settings/security-settings.asciidoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (albertzaharovits) [reply docs/reference/settings/security-settings.asciidoc:241]: Thanks for the feedback folks!
I'll go with keeping this documented, and added a NOTE that this controls the per-node deletion interval, not a per-cluster one. [48bc1c8](https://github.com/elastic/elasticsearch/pull/105392/commits/48bc1c85769f38a1ed86ea294b4de33e430b754b)

File: docs/reference/settings/security-settings.asciidoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (albertzaharovits) [reply docs/reference/settings/security-settings.asciidoc:248]: Thanks for the feedback! Documenting it seems like the right choice!

**Pull request merged**